src/Microsoft.AspNet.Mvc.Razor/Compilation/RoslynCompilationService.cs

```diff
@@ -256,5 +247,21 @@ private static bool SupportsPdbGeneration()
 
             return false;
         }
+
+        private static string ReadFileContentsSafely(IFileInfo fileInfo)
```

Comment:
Consider making this a static method on one of the compilation result or error classes. Right now it's duped :sob: